wandb/sdk/internal/job_builder.py

```diff
@@ -167,7 +167,10 @@ def _build_repo_job(
             },
         }
 
-        name = make_artifact_name_safe(f"job-{remote}_{program_relpath}")
+        if hasattr(self._settings, "job_name") and self._settings.job_name:
```

Comment:
this is sort of a strange pattern, when do runs not have the attr at all?

Reply:
offline mode can cause that, if the static settings are not set right

Reply:
the settings objects attributes are created dynamically such that if the user hasn't set the argument then the settings object here will not have the attribute. 🤷🏻‍♂️

Reply:
Looks like @dmitryduev did a settings refactor today so I'm not sure if this is still true.

Reply:
I think this is no longer necessary thanks to Dima's change

Reply:
+1 to @KyleGoyette, `self._settings.job_name` should just work (it's a bug if it doesn't).